addons/frontend/src/svg-patch.ts

```diff
@@ -332,8 +332,11 @@ export function patchRoot(prev: SVGElement, next: SVGElement) {
   /// Patch global svg resources
   patchSvgHeader(prev, next);
 
+  /// Hard replace elements that is not a `<g>` element.
+  const frozen = preReplaceNonSVGElements(prev, next, 3);
   /// Patch `<g>` children, call `reuseOrPatchElem` to patch.
   patchChildren(prev, next);
+  postReplaceNonSVGElements(prev, 3, frozen);
```

Comment:
can you explain a bit about the magic number `3` here?

Reply:
Please see https://github.com/Enter-tainer/typst-preview/pull/120/commits/ba84d5079c66f2c594c449c69d1892a73a8efca9#diff-aa4f1a91aeb50ae8eb47dc4b1bb2f2322446c5dd5af2c1493b924433b16c2496R491

I have actively refactored some magic code and add some comments about this.